go/libraries/doltcore/diff/diff_summary.go

```diff
@@ -86,3 +120,40 @@ func reportChanges(ctx context.Context, change *diff.Difference, ch chan<- DiffS
 		return ctx.Err()
 	}
 }
+
+func reportKeylessChanges(ctx context.Context, change *diff.Difference, ch chan<- DiffSummaryProgress) error {
+	var oldCard uint64
+	if change.OldValue != nil {
+		v, err := change.OldValue.(types.Tuple).Get(row.KeylessCardinalityValIdx)
+		if err != nil {
+			return err
+		}
+		oldCard = uint64(v.(types.Uint))
+	}
+
+	var newCard uint64
+	if change.NewValue != nil {
+		v, err := change.NewValue.(types.Tuple).Get(row.KeylessCardinalityValIdx)
+		if err != nil {
+			return err
+		}
+		newCard = uint64(v.(types.Uint))
+	}
+
+	var summary DiffSummaryProgress
+	delta := int64(newCard) - int64(oldCard)
+	if delta > 0 {
+		summary = DiffSummaryProgress{Adds: uint64(delta)}
+	} else if delta < 0 {
+		summary = DiffSummaryProgress{Removes: uint64(-delta)}
+	} else {
+		return fmt.Errorf("diff with delta = 0 for key: %s", change.KeyValue.HumanReadableString())
+	}
+
+	select {
```

Comment:
Is this idiomatic / kosher? Never seen sending and receiving on channels in the same select block before

(for that matter I don't think I've ever seen sending in a select, at all)

Reply:
This was copied from AsyncDiffer, don't @ me